src/engraving/rendering/dev/arpeggiolayout.cpp

```diff
@@ -33,25 +37,291 @@ using namespace mu::engraving::rendering::dev;
+    if (!endChord) {
+        endChord = item->chord();
+    }
+    const Part* part = item->part();
+    const track_idx_t partStartTrack = part->startTrack();
+    const track_idx_t partEndTrack = part->endTrack();
+    const PaddingTable paddingTable = item->score()->paddingTable();
+
+    double accidentalDistance = paddingTable.at(ElementType::ARPEGGIO).at(ElementType::ACCIDENTAL) * item->mag();
+    double arpeggioLedgerDistance = paddingTable.at(ElementType::ARPEGGIO).at(ElementType::LEDGER_LINE) * item->mag();
+
+    Shape arpShape = item->shape().translate(item->pagePos());
+    double arpX = arpShape.right();
+    double largestOverlap = 0.0;
+
+    for (track_idx_t track = partStartTrack; track < partEndTrack; ++track) {
+        EngravingItem* e = seg->element(track);
+        if (!e || !e->isChord()) {
+            continue;
+        }
+
+        Chord* chord = toChord(e);
+        // Only check chords the arpeggio doesn't span on staves the arpeggios span
+        // We already calculated the gap between chords it does span in ChordLayout::layoutPitched
+        bool aboveStart
+            = std::make_pair(chord->vStaffIdx(), chord->downLine()) < std::make_pair(item->vStaffIdx(), item->chord()->upLine());
+        bool belowEnd = std::make_pair(chord->vStaffIdx(), chord->upLine()) > std::make_pair(endChord->vStaffIdx(), endChord->downLine());
+
+        if ((chord->spanArpeggio() == item && !(aboveStart || belowEnd)) || chord->vStaffIdx() < item->vStaffIdx()
+            || chord->vStaffIdx() > endChord->vStaffIdx()) {
+            continue;
+        }
+
+        // We don't know the distance between staves, so our arpeggio's height is currently unknown if it spans staves
+        // To make sure we spot collisions with non-arpeggio accidentals, extend arpeggio upwards/downwards a large amount
+        // depending on if they are the end/start of the arpeggio respectively.
+        // This is to find collisions between chords outside the arpeggios track span, but within its "visual span"
+        Shape curArpShape = arpShape;
+        staff_idx_t staffIdx = chord->vStaffIdx();
+        if (staffIdx == item->chord()->vStaffIdx() && item->crossStaff()) {
+            // Arpeggio doesn't end on this staff, so extend downwards
+            curArpShape.addBBox(RectF(curArpShape.bbox().bottomLeft().x(), curArpShape.bbox().bottomLeft().y(),
+                                      curArpShape.bbox().width(), 10000));
+        } else if (staffIdx == endChord->vStaffIdx() && item->crossStaff()) {
+            // Arpeggio doesn't start on this staff so move to last note and extend upwards
+            curArpShape.translate(PointF(0.0, endChord->downNote()->pagePos().y() - item->pagePos().y()));
+            curArpShape.addBBox(RectF(curArpShape.bbox().topLeft().x(), curArpShape.bbox().topLeft().y() - 10000,
+                                      curArpShape.bbox().width(), 10000));
```

Comment:
Please use a named constant instead of this magic number (same for other places where you use it)